### ISSUE TITLE
Error running the patcher

### ISSUE THREAD

**Issue by scoufman** [stoneharry/RCEPatcher#1]
Hi,

I get the following error when I try to run the program:

Cannot use file stream for [C:\wow\RCEPatcher.deps.json]: No such file or directory
A fatal error was encountered. The library 'hostpolicy.dll' required to execute the application was not found in 'C:\Program Files\dotnet\'.
Failed to run as a self-contained app.
  - The application was run as a self-contained app because 'C:\wow\RCEPatcher.runtimeconfig.json' was not found.
  - If this should be a framework-dependent app, add the 'C:\wow\RCEPatcher.runtimeconfig.json' file and specify the appropriate framework.

Any clue on what might be missing?

Thanks in advance.

**Comment by bladerita** [comment]
https://dotnet.microsoft.com/download/dotnet/thank-you/runtime-desktop-5.0.10-windows-x64-installer
https://dotnet.microsoft.com/download/dotnet-framework/thank-you/net47-offline-installer
https://learn.microsoft.com/ru-ru/cpp/windows/latest-supported-vc-redist?view=msvc-170
https://learn.microsoft.com/ru-ru/windows/apps/windows-app-sdk/downloads

**Comment by stoneharry** [comment]
[RCEPatcher_1_1.zip](https://github.com/user-attachments/files/16404897/RCEPatcher_1_1.zip)

This should fix the error. .NET is required, and it should prompt to download with this version.

**Comment by scoufman** [comment]
Thank you, installed the 8.0.7 version (https://download.visualstudio.microsoft.com/download/pr/3980ab0a-379f-44a0-9be6-eaf74c07a3b3/bd1cc6107ff3d8fe0104d30f01339b74/dotnet-runtime-8.0.7-win-x64.exe) and it ran properly afterwards.

**Comment by stoneharry** [comment]
Reopening this to track downgrading the .NET version to one that most people will have. It does not need to be on .NET 8.

I should also make it pause on input when an error occurs.

**Comment by THE-GOD-OF-WORLD-OF-WARCRAFT** [comment]
any patch for Shenna-crack-wow 1.12.1  ?

**Comment by stoneharry** [comment]
> any patch for Shenna-crack-wow 1.12.1 ? or whatever version turtle wow runs

1.12 and 2.4.3 Blizzard's private key is cracked (due to using a weaker number of bits) so servers can send any content and sign it as being safe, so the client just accepts it. So there is no point exploring to see if the same RCE exists in earlier versions, you can already completely bypass any protections the client has.

**Comment by THE-GOD-OF-WORLD-OF-WARCRAFT** [comment]
> > any patch for Shenna-crack-wow 1.12.1 ? or whatever version turtle wow runs
> 
> 1.12 and 2.4.3 Blizzard's private key is cracked (due to using a weaker number of bits) so servers can send any content and sign it as being safe, so the client just accepts it. So there is no point exploring to see if the same RCE exists in earlier versions, you can already completely bypass any protections the client has.

thank you :)

**Comment by stoneharry** [comment]
Published v1.2 to help here.

**Comment by temzky** [comment]
> Reopening this to track downgrading the .NET version to one that most people will have. It does not need to be on .NET 8.

.NET 5 support ended over 2 years ago, It's better to use .NET LTS (long term support) release
.NET 5 is not included with Windows, and it seems it's not forward compatible:
```
You must install or update .NET to run this application.

App: D:\Games\World of Warcraft 3.3.5a\RCEPatcher\RCEPatcher.exe
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '5.0.0' (x64)
.NET location: C:\Program Files\dotnet\

The following frameworks were found:
  8.0.6 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  8.0.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=5.0.0&arch=x64&rid=win-x64&os=win10
```
